tools/xtu-build-new/xtu-build.py

```diff
@@ -218,48 +266,51 @@ def map_functions(command):
 
 # Generate externalFnMap.txt
 
-func_2_file = {}
-func_2_size = {}
-extfunc_2_file = {}
-func_2_fileset = {}
-
-defined_fns_filename = os.path.join(mainargs.xtuindir, 'definedFns.txt')
-with open(defined_fns_filename,  'r') as defined_fns_file:
-    for line in defined_fns_file:
-        funcname, filename, funlen = line.strip().split(' ')
-        if funcname.startswith('!'):
-            funcname = funcname[1:]  # main function
-        if funcname not in func_2_file.keys():
-            func_2_fileset[funcname] = set([filename])
-        else:
-            func_2_fileset[funcname].add(filename)
-        func_2_file[funcname] = filename
-        func_2_size[funcname] = funlen
-
-extern_fns_filename = os.path.join(mainargs.xtuindir, 'externalFns.txt')
-with open(extern_fns_filename,  'r') as extern_fns_file:
-    for line in extern_fns_file:
-        line = line.strip()
-        if line in func_2_file and line not in extfunc_2_file:
-            extfunc_2_file[line] = func_2_file[line]
-
-extern_fns_map_filename = os.path.join(mainargs.xtuindir, 'externalFnMap.txt')
-with open(extern_fns_map_filename, 'w') as out_file:
-    for func, fname in extfunc_2_file.items():
-        if len(func_2_fileset[func]) == 1:
-            if not mainargs.reparse:
-                out_file.write('%s %s.ast\n' % (func, fname))
-            else:
-                out_file.write('%s %s\n' % (func, fname))
+create_external_fn_maps(mainargs.xtuindir)
+shutil.rmtree(os.path.join(mainargs.xtuindir, TEMP_EXTERNAL_FNMAP_FOLDER), ignore_errors=True)
+
+# func_2_file = {}
+# func_2_size = {}
+# extfunc_2_file = {}
+# func_2_fileset = {}
+
+# defined_fns_filename = os.path.join(mainargs.xtuindir, 'definedFns.txt')
```

Comment:
It seemed easier to leave this code for experimenting because this whole script will be deleted after a while. However, I remove it then.